integration/spark-common/src/main/scala/org/apache/spark/sql/optimizer/CarbonDecoderOptimizerHelper.scala

```diff
@@ -84,7 +98,16 @@ class CarbonDecoderProcessor {
         }
         nodeList.add(ArrayCarbonNode(nodeListSeq))
       case e: UnaryNode => process(e.child, nodeList)
-      case i: InsertIntoTable => process(i.child, nodeList)
+      case i: InsertIntoTable =>
+        var sparkVersion21: Boolean = false
+
+        sparkVersion21 = !CarbonReflectionUtils.hasField("query", InsertIntoTable)
```

Comment:
Based on the version availability from spark check the version not depends on the field.